### PR TITLE
New BETA

### PR DESCRIPTION
**Pull request by Raycoms**
Things were working well, let's make a new beta.

# Changes proposed in this pull request:
-
-
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/CitizenData.java [hunk 1]
```diff
@@ -232,6 +232,70 @@ public void markDirty()
         colony.markCitizensDirty();
     }
 
+    @Override
+    public boolean equals(final Object o)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/CitizenData.java:236]: MAJOR Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 2]
```diff
@@ -40,7 +43,12 @@
 
 /**
  * Base building class, has all the foundation for what a building stores and does.
+ *
+ * We suppress the warning which warns you about referencing child classes in the parent because that's how we register the instances of the childClasses
+ * to their views and blocks.
+ *
  */
+@SuppressWarnings("squid:S2390")
 public abstract class AbstractBuilding
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java:52]: MAJOR Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 37 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
MAJOR class "AbstractBuilding" has 66 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 3]
```diff
@@ -171,84 +157,92 @@ public IBlockState getStateFromMeta(final int meta)
...
     }
 
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
-    public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:243]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 4]
```diff
@@ -171,84 +157,92 @@ public IBlockState getStateFromMeta(final int meta)
...
 
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SuppressWarnings(DEPRECATION)
+    @Nullable
+    @Deprecated
     @Override
-    public boolean isPassable(final IBlockAccess worldIn, final BlockPos pos)
+    public AxisAlignedBB getCollisionBoundingBox(
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:231]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 5]
```diff
@@ -171,84 +157,92 @@ public IBlockState getStateFromMeta(final int meta)
...
 
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
+    @SideOnly(Side.CLIENT)
     @Override
     @Deprecated
-    public boolean isFullCube(final IBlockState state)
+    public boolean shouldSideBeRendered(final IBlockState blockState, final IBlockAccess blockAccess, final BlockPos pos, final EnumFacing side)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:219]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 6]
```diff
@@ -171,84 +157,92 @@ public IBlockState getStateFromMeta(final int meta)
...
     }
 
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
-    public boolean isOpaqueCube(final IBlockState state)
+    public boolean isFullCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:172]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 7]
```diff
@@ -171,84 +157,92 @@ public IBlockState getStateFromMeta(final int meta)
...
-    @NotNull
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SuppressWarnings(DEPRECATION)
     @Override
-    @SideOnly(Side.CLIENT)
-    public BlockRenderLayer getBlockLayer()
+    @Deprecated
+    public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:189]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 8]
```diff
@@ -1479,7 +1507,7 @@ public void spawnCitizen(final CitizenData data)
             entity.setColony(this, citizenData);
 
             entity.setPosition(spawnPoint.getX() + 0.5D, spawnPoint.getY() + 0.1D, spawnPoint.getZ() + 0.5D);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java:1509]: MAJOR Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 0.1D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 9]
```diff
@@ -44,16 +47,15 @@
...
 import java.util.stream.Collectors;
 
-import static com.minecolonies.api.util.constant.Constants.SECONDS_A_MINUTE;
-import static com.minecolonies.api.util.constant.Constants.TICKS_SECOND;
+import static com.minecolonies.api.util.constant.Constants.*;
 
 /**
  * This class describes a colony and contains all the data and methods for manipulating a Colony.
  */
 public class Colony implements IColony
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java:55]: MAJOR Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 48 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
MAJOR class "Colony" has 93 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 10]
```diff
@@ -4,6 +4,7 @@
 import com.minecolonies.api.colony.permissions.Rank;
 import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.*;
+import com.minecolonies.api.util.constant.Constants;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java:7]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Constants'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 11]
```diff
@@ -870,6 +883,17 @@ public void onDeath(final DamageSource par1DamageSource)
...
     }
 
+    @Override
+    public EnumActionResult applyPlayerInteraction(
+            final EntityPlayer player,
+            final Vec3d vec,
+            @Nullable final ItemStack itemInHand,
+            final EnumHand hand)
+    {
+        SoundUtils.playInteractionSoundAtCitizenWithChance(CompatibilityUtils.getWorld(this), this.getPosition(), 100, this);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:893]: MAJOR Assign this magic number 100 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java [hunk 12]
```diff
@@ -101,7 +101,7 @@ public void renderWorldLastEvent(@NotNull final RenderWorldLastEvent event)
             }
 
             final ColonyView colony = ColonyManager.getClosestColonyView(world, player.getPosition());
-            if(colony != null && colony.getPermissions().hasPermission(player, Action.ACCESS_HUTS))
+            if(colony != null && player != null && colony.getPermissions().hasPermission(player, Action.ACCESS_HUTS))
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java:104]: MAJOR Remove this expression which always evaluates to "true" [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)

File: src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java [hunk 13]
```diff
@@ -947,7 +947,7 @@ public boolean hasPermission(final UUID id, @NotNull final Action action)
         public boolean hasPermission(final Rank rank, @NotNull final Action action)
         {
             return (rank == Rank.OWNER && action != Action.GUARDS_ATTACK)
-                     || Utils.testFlag(permissions.get(rank), action.getFlag());
+                     || (permissions != null && action != null && Utils.testFlag(permissions.get(rank), action.getFlag()));
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java:950]: MAJOR Remove this expression which always evaluates to "true" [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java [hunk 14]
```diff
@@ -324,7 +323,8 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
 
         // If we have no active worker, grab one from the Colony
         // TODO Maybe the Colony should assign jobs out, instead?
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:325]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/blockout/PaneParams.java [hunk 15]
```diff
@@ -432,16 +432,16 @@ else if ((attr.startsWith("rgb(") || attr.startsWith("rgba(")) && m.find())
...
     private static int getRGBA(final String attr, final Matcher m)
     {
-        final int r = MathHelper.clamp_int(Integer.parseInt(m.group(1)), 0, 255);
-        final int g = MathHelper.clamp_int(Integer.parseInt(m.group(2)), 0, 255);
-        final int b = MathHelper.clamp_int(Integer.parseInt(m.group(3)), 0, 255);
+        final int r = MathHelper.clamp(Integer.parseInt(m.group(1)), 0, 255);
+        final int g = MathHelper.clamp(Integer.parseInt(m.group(2)), 0, 255);
+        final int b = MathHelper.clamp(Integer.parseInt(m.group(3)), 0, 255);
 
         int color = ((r & 0xFF) << 16) | ((g & 0xFF) << 8) | (b & 0xFF);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/blockout/PaneParams.java:439]: MAJOR Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 8 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/blockout/PaneParams.java [hunk 16]
```diff
@@ -432,16 +432,16 @@ else if ((attr.startsWith("rgb(") || attr.startsWith("rgba(")) && m.find())
...
 
         if (attr.startsWith("rgba"))
         {
             final int alpha = (int) (Double.parseDouble(m.group(4)) * 255.0F);
-            color |= MathHelper.clamp_int(alpha, 0, 255) << 24;
+            color |= MathHelper.clamp(alpha, 0, 255) << 24;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/blockout/PaneParams.java:444]: MAJOR Assign this magic number 255 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 24 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/blockout/PaneParams.java [hunk 17]
```diff
@@ -346,7 +346,7 @@ private static int parseScalableIntegerRegexMatch(final Matcher m, final int def
 
             if ("%".equals(m.group(2)))
             {
-                value = scale * MathHelper.clamp_int(value, 0, 100) / 100;
+                value = scale * MathHelper.clamp(value, 0, 100) / 100;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/blockout/PaneParams.java:349]: MAJOR Assign this magic number 100 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 100 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java [hunk 18]
```diff
@@ -164,6 +164,7 @@ public void removeCitizen(@NotNull final CitizenData citizen)
         {
             citizen.setHomeBuilding(null);
             residents.remove(citizen);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:166]: MINOR This call to "remove()" may be a performance hot spot if the collection is large. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java [hunk 19]
```diff
@@ -132,20 +133,21 @@ public boolean isPassable(final IBlockAccess worldIn, final BlockPos pos)
...
-                                  (float) HEIGHT_COLLISION,
-                                  (float) END_COLLISION);
+                (float) BOTTOM_COLLISION,
+                (float) START_COLLISION,
+                (float) END_COLLISION,
+                (float) HEIGHT_COLLISION,
+                (float) END_COLLISION);
     }
 
     //todo: remove once we no longer need to support this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java:149]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java [hunk 20]
```diff
@@ -132,20 +133,21 @@ public boolean isPassable(final IBlockAccess worldIn, final BlockPos pos)
...
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java:135]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java [hunk 21]
```diff
@@ -118,7 +119,7 @@ public int getMetaFromState(@NotNull final IBlockState state)
...
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java:121]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 22]
```diff
@@ -284,56 +229,110 @@ public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess
...
-    // =======================================================================
-    // ======================= Rendering & IBlockState =======================
-    // =======================================================================
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SideOnly(Side.CLIENT)
+    @Override
+    @Deprecated
+    public boolean shouldSideBeRendered(final IBlockState blockState, final IBlockAccess blockAccess, final BlockPos pos, final EnumFacing side)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:273]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 23]
```diff
@@ -284,56 +229,110 @@ public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess
...
+    {
+        return null;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:295]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 24]
```diff
@@ -284,56 +229,110 @@ public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess
...
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Deprecated
+    @Nullable
+    @Override
+    public AxisAlignedBB getCollisionBoundingBox(
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:284]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/PathNavigate.java [hunk 25]
```diff
@@ -90,7 +90,7 @@ public BlockPos getDestination()
     protected boolean isDirectPathBetweenPoints(final Vec3d start, final Vec3d end, final int sizeX, final int sizeY, final int sizeZ)
     {
         // TODO improve road walking. This is better in some situations, but still not great.
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/PathNavigate.java:92]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java [hunk 26]
```diff
@@ -89,6 +93,7 @@ public TileEntity createNewTileEntity(final World world, final int meta)
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
     public boolean isFullBlock(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java:99]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java [hunk 27]
```diff
@@ -166,6 +175,7 @@ public boolean isFullCube(final IBlockState state)
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
     public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java:181]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

**Comment by MineShadow7** [comment]
I thought you wanted to make it BETA after release all planned workers.

**Comment by Raycoms** [comment]
Beta is just the stage in curse

**Pull request merged**